modules/ROOT/pages/traversal-framework/traversal_framework_example.adoc

```diff
@@ -0,0 +1,220 @@
+
+the graph can be traversed with, for example, the following traverser starting at the node with the `name = 'Joe'`:
+
+[source, java]
+----
+for ( Path position : db.traversalDescription()
+        .depthFirst()
+        .relationships( Rels.KNOWS )
+        .relationships( Rels.LIKES, Direction.INCOMING )
+        .evaluator( Evaluators.toDepth( 5 ) )
+        .traverse( node ) )
+{
+    output += position + "\n";
+}
+----
+
+The traversal will thus output:
+
+[source, output, role="noheader"]
+----
+(0)
+(0)<-[LIKES,1]-(5)
+(0)<-[LIKES,1]-(5)-[KNOWS,6]->(1)
+(0)<-[LIKES,1]-(5)-[KNOWS,6]->(1)<-[KNOWS,5]-(6)
+(0)<-[LIKES,1]-(5)-[KNOWS,6]->(1)-[KNOWS,4]->(4)
+(0)<-[LIKES,1]-(5)-[KNOWS,6]->(1)-[KNOWS,4]->(4)-[KNOWS,3]->(3)
+(0)<-[LIKES,1]-(5)-[KNOWS,6]->(1)-[KNOWS,4]->(4)-[KNOWS,3]->(3)-[KNOWS,2]->(2)
+----
+
+Since a `TraversalDescription` is immutable, it is useful to create template descriptions that include common settings shared by different traversals.
+For example, start with this traverser:
+
+[source, java]
+----
+friendsTraversal = db.traversalDescription()
+        .depthFirst()
+        .relationships( Rels.KNOWS )
+        .uniqueness( Uniqueness.RELATIONSHIP_GLOBAL );
+----
+
+
+It should yield the following output (starting from the node with the `name = 'Joe'`):
+
+[source, java]
+----
+(0)
+(0)-[KNOWS,0]->(2)
+(0)-[KNOWS,0]->(2)<-[KNOWS,2]-(3)
+(0)-[KNOWS,0]->(2)<-[KNOWS,2]-(3)<-[KNOWS,3]-(4)
+(0)-[KNOWS,0]->(2)<-[KNOWS,2]-(3)<-[KNOWS,3]-(4)<-[KNOWS,4]-(1)
+(0)-[KNOWS,0]->(2)<-[KNOWS,2]-(3)<-[KNOWS,3]-(4)<-[KNOWS,4]-(1)<-[KNOWS,6]-(5)
+(0)-[KNOWS,0]->(2)<-[KNOWS,2]-(3)<-[KNOWS,3]-(4)<-[KNOWS,4]-(1)<-[KNOWS,5]-(6)
+----
+
+Then create a new traverser from it, restricting depth to three:
+
+[source, java]
+----
+for ( Path path : friendsTraversal
+        .evaluator( Evaluators.toDepth( 3 ) )
+        .traverse( node ) )
+{
+    output += path + "\n";
+}
+----
+
+This should be the output:
+
+[source, output, role="noheader"]
+----
+(0)
+(0)-[KNOWS,0]->(2)
+(0)-[KNOWS,0]->(2)<-[KNOWS,2]-(3)
+(0)-[KNOWS,0]->(2)<-[KNOWS,2]-(3)<-[KNOWS,3]-(4)
+----
+
+
+In case you want to traverse from depth two to four:
+
+[source, java]
+----
+for ( Path path : friendsTraversal
+        .evaluator( Evaluators.fromDepth( 2 ) )
+        .evaluator( Evaluators.toDepth( 4 ) )
+        .traverse( node ) )
+{
+    output += path + "\n";
+}
+----
+
+This will give the following output:
+
+[source, output, role="noheader"]
+----
+(0)-[KNOWS,0]->(2)<-[KNOWS,2]-(3)
+(0)-[KNOWS,0]->(2)<-[KNOWS,2]-(3)<-[KNOWS,3]-(4)
+(0)-[KNOWS,0]->(2)<-[KNOWS,2]-(3)<-[KNOWS,3]-(4)<-[KNOWS,4]-(1)
+----
+
+To see other useful evaluators, check the link:{org-neo4j-graphdb-traversal-Evaluators}[`org.neo4j.graphdb.traversal.Evaluators`^] Java API or simply implement the link:{org-neo4j-graphdb-traversal-Evaluator}[`org.neo4j.graphdb.traversal.Evaluator`^] interface yourself.
+
+The `Traverser` also has a `nodes()` method that will return an iterable of all the nodes in the paths:
```

Comment:
An iterable what? Iterable list, result?

Reply:
The Java Iterable (https://docs.oracle.com/javase/8/docs/api/java/lang/Iterable.html), I capitalised it and added back ticks around it to show it is part of the code now, I think that helps :)

Reply:
It does! Thank you :)